earthaccess/results.py

```diff
@@ -172,6 +175,46 @@ def s3_bucket(self) -> Dict[str, Any]:
             return self["umm"]["DirectDistributionInformation"]
         return {}
 
+    def services(self) -> Dict[Any, List[Dict[str, Any]]]:
+        """
+        Returns:
+            A list of services available for the collection.
+        """
+
+        services = self.get("meta", {}).get("associations", {}).get("services", [])
+
+        parsed = {}
+        for service in services:
+            if earthaccess.__auth__.authenticated:
+                query = DataService(auth=earthaccess.__auth__).parameters(
+                    concept_id=service
+                )
+            else:
+                query = DataService().parameters(concept_id=service)
+            results = query.get(query.hits())
+            parsed[service] = self._parse_service_result(results)
```

Comment:
Once you make the change I suggested in the `get` method, you should be able to eliminate the `_parse_service_result` method and just do this:
```suggestion
            parsed[service] = query.get(query.hits())
```